### PR TITLE
Treatment date range bug fix

### PR DESCRIPTION
**Pull request by cvalarida**
## Description
This PR:
1) Adds a validation to require a month and day _if_ anything is in the date field  
2) Uses that validator on MVP and v2
3) Sets the `day` to the last day of the month in the submit transformer
    - In MVP and v2

**Note:** The validation message didn't show up until the continue button is hit on the review page. Not sure what the deal is there, but I don't _think_ it's part of this PR.

## Testing done
Unit tests written for the validator.
Manually tested that the translation worked in MVP and v2.

## Screenshots
**V2**
image

**MVP**
image

## Acceptance criteria
- [x] The treatment date range end date requires a month and year if any information is entered
- [x] The treatment date range end date is set to the last of the month

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/disability-benefits/all-claims/utils.jsx [hunk 1]
```diff
@@ -346,18 +346,32 @@ export function transform(formConfig, form) {
...
+      );
+
+      // If the day is missing, set it to the last day of the month because EVSS requires a day
+      const [year, month, day] = _.get(
+        'treatmentDateRange.to',
+        newFacility,
+        '',
+      ).split('-');
+      if (day && day === 'XX') {
+        newFacility.treatmentDateRange.to = moment(`${year}-${month}`)
```
Comment (cvalarida) [inline src/applications/disability-benefits/all-claims/utils.jsx]: Should this be applied to the `from` date as well? We're sending `XX` days on that one.

File: src/applications/disability-benefits/all-claims/utils.jsx [hunk 2]
```diff
@@ -346,18 +346,32 @@ export function transform(formConfig, form) {
...
         transformRelatedDisabilities(
           facility.treatedDisabilityNames,
           claimedConditions,
         ),
         facility,
-      ),
-    );
+      );
+
+      // If the day is missing, set it to the last day of the month because EVSS requires a day
```
Comment (cvalarida) [inline src/applications/disability-benefits/all-claims/utils.jsx]: This whole chunk could be abstracted into a function and shared between MVP and all claims. Needs some tweaking for that to work, though.

**Review by erikphansen — APPROVED** [review]
LGTM. @cvalarida can decide if he needs someone with more 526-specific knowledge to review this before merging or not!

**Review by markgreenburg — APPROVED** [review]
lgtm

**Pull request merged**